threediresultstyler/threediresultstyler_dockwidget.py

```diff
@@ -0,0 +1,259 @@
+        pdok_layer.setCustomProperty("embeddedWidgets/count", 1)
+        pdok_layer.setCustomProperty("embeddedWidgets/0/id", "transparency")
+        self.iface.layerTreeView().refreshLayerSymbology(pdok_layer.id())
+        
+    def save_tree_settings(self):
+        root = QgsProject.instance().layerTreeRoot()
+        schematisation_group = root.findGroup('schematisation')
+        groups = ['settings','boundary conditions','laterals','1d','additional tables for editing 1d','obstacles','grid refinements','advanced numerics','inflow','impervious_surface','surface']
+ 
+        settings = {}
+        settings['group'] = {}
+        settings['layer'] = {}
+
+        for group_name in groups:
+            group = schematisation_group.findGroup(group_name)
+
+            settings['group'][group_name] = {}
+            settings['group'][group_name]['visible'] = group.isVisible()
+            settings['group'][group_name]['expanded'] = group.isExpanded()
+
+        for layer in schematisation_group.findLayers():
+            settings['layer'][layer.name()] = {}
+            settings['layer'][layer.name()]['visible'] = layer.isVisible()
+            settings['layer'][layer.name()]['expanded'] = layer.isExpanded()
+            settings['layer'][layer.name()]['subset'] = layer.layer().subsetString()
+
+        settings_dir = os.path.join(os.path.dirname(__file__), 'tree_settings')
+        filename = QtGui.QFileDialog.getSaveFileName(self, 'Open file', settings_dir,"JSON files (*.json)")[0]
+
+
+        with open(str(filename), 'w', encoding='utf-8') as outfile:
+            json.dump(settings, outfile, ensure_ascii=False, indent=2)
+            
+    def load_tree_settings(self):
+        settings_dir = os.path.join(os.path.dirname(__file__), 'tree_settings')
+        filename = QtGui.QFileDialog.getOpenFileName(self, 'Open file', settings_dir,"JSON files (*.json)")[0]
+        
+        root = QgsProject.instance().layerTreeRoot()
+        schematisation_group = root.findGroup('schematisation')
+        
+        with open(filename) as json_file:  
+            settings = json.load(json_file)
+
+        groups = schematisation_group.findGroups()
+        layers = schematisation_group.findLayers()
+        
+        for group in groups:
+            try:
+                group.setItemVisibilityChecked(settings['group'][group.name()]['visible'])
+                group.setExpanded(settings['group'][group.name()]['expanded'])
+            except:
+                self.iface.messageBar().pushMessage("Warning", "Couldn't find settings for group {}".format(group.name()), level=Qgis.Warning)
+            
+        for layer in layers:
+            try:
+                layer.setItemVisibilityChecked(settings['layer'][layer.name()]['visible'])
+                layer.setExpanded(settings['layer'][layer.name()]['expanded'])
+                layer.layer().setSubsetString(settings['layer'][layer.name()]['subset'])
+            except:
```

Comment:
Dit is een "kale except": die vangt alles af, inclusief typfoutjes in `'layer'`, bijvoorbeeld. Alles.
Dus misschien even kijken welke Exception je precies af wilt vangen?

Reply:
fixed